### PR TITLE
Tidy up compatible MCUs docs

### PR DESCRIPTION
**Pull request by fauxpark**
## Description
Made this page a little more comprehensive, and actually added it to the nav.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/compatible_microcontrollers.md [hunk 1]
```diff
@@ -1,25 +1,34 @@
...
+* [ATmega328P](https://www.microchip.com/wwwproducts/en/ATmega328P)
 
-You can also use any ARM processor that [ChibiOS](http://www.chibios.org) supports. The following processors have been tested:
+## ARM
 
-* [Kinetis MKL26Z64](http://www.nxp.com/products/microcontrollers-and-processors/arm-processors/kinetis-cortex-m-mcus/l-series-ultra-low-power-m0-plus/kinetis-kl2x-48-mhz-usb-ultra-low-power-microcontrollers-mcus-based-on-arm-cortex-m0-plus-core:KL2x)
-* [Kinetis MK20DX128](http://www.nxp.com/assets/documents/data/en/data-sheets/K20P64M50SF0.pdf)
-* [Kinetis MK20DX128](http://www.nxp.com/assets/documents/data/en/data-sheets/K20P64M50SF0.pdf)
-* [Kinetis MK20DX256](http://www.nxp.com/products/microcontrollers-and-processors/arm-processors/kinetis-cortex-m-mcus/k-series-performance-m4/k2x-usb/kinetis-k20-72-mhz-full-speed-usb-mixed-signal-integration-microcontrollers-mcus-based-on-arm-cortex-m4-core:K20_72)
+You can also use any ARM chip that [ChibiOS](http://www.chibios.org) supports. These have plenty of flash. Known to work are:
```
Comment (zvecr) [inline docs/compatible_microcontrollers.md]: Not all ST chips have USB. Something like a f042 could run QMK, but the 16KB version will really struggle.
```suggestion
You can also use any ARM chip that [ChibiOS](http://www.chibios.org) supports that has USB. Most have plenty of flash. Known to work are:
```

Comment (fauxpark) [reply docs/compatible_microcontrollers.md]: There are two F042 boards already in the repo, in fact. One of them calls out the STM32F042K6T6 specifically, which has 32KB.

Comment (zvecr) [reply docs/compatible_microcontrollers.md]: Different variants can have different amounts of flash, STM32F042K4 is 16KB.
STM32F030 runs chibios, but has no USB so kinda useless in the current QMK setup.

Comment (fauxpark) [reply docs/compatible_microcontrollers.md]: Could maybe say that 32KB is the absolute minimum for STM32/ARM/both? Move the sentence that talks about it in the AVR section further up to an intro paragraph.

**Pull request merged**